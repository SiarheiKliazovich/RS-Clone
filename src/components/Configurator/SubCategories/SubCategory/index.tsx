import { FunctionComponent } from "react";
import { SubCategoryType } from "../../../../types/configurator";
import { nightTheme } from "../../../../data/constants";

const SubCategory: FunctionComponent<SubCategoryType> = ({
  setType,
  subCategory,
  accentColor,
  isNavbarNightMode,
}: SubCategoryType) => {
  const backgroundColor = nightTheme.background.element;

  return (
    <li
      className="image-item"
      key={subCategory.id}
      onClick={() => setType(subCategory.type)}
    >
      <div className="img-wrapper">
        <img className="img-subcategories" src={subCategory.image} alt="img" />
      </div>
      <div className="btn-wrapper">
        <button
          style={{
            backgroundColor: isNavbarNightMode ? backgroundColor : accentColor,
          }}
          className="category-name"
        >
          {subCategory.name}
        </button>
      </div>
    </li>
  );
};
export default SubCategory;
