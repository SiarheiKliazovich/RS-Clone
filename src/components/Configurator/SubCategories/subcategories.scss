@import "../../../sass/variables";
@import "../../../sass/mixins";


.subcategories-wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 40px 0px;
  background: $color-white1;
}

.title-subcategories {
  font-size: 22px;
  line-height: 33px;
  font-weight: 300;
  text-transform: capitalize;
  color: $color-black1;
  padding-bottom: 15px;
}

.images-list {
  list-style: none;
  display: flex;
  padding: 0;
  width: 100%;
  margin: 0;
  justify-content: center;
  flex-wrap: wrap;
  gap: 0 10px;

  @include media-small-mobile {
    flex-direction: column;
    align-items: center;
  }
}

.image-item {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 20px;
  align-items: center;
  background: $color-white1;
  border-radius: 5px;

  .btn-wrapper {
    margin-bottom: 10px;
  }
}

.tie-tipe {
  background: $color-grey3;
  padding: 3px 30px;
  border-radius: 5px;
}

.img-subcategories {
  border-radius: 20px;
}
