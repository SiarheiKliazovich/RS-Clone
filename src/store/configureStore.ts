/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
