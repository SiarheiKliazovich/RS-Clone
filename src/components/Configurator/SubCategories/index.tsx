import { FunctionComponent } from "react";
import "./subcategories.scss";
import { ISubcategory } from "../../../interfaces/configurator";
import { SubCategoriesType } from "../../../types/configurator";
import View from "../View";
import SubCategory from "./SubCategory";

const SubCategories: FunctionComponent<SubCategoriesType> = ({
  type,
  setType,
  subCategories,
}: SubCategoriesType) => (
  <div className="subcategories-wrapper">
    <h4 className="title">Tie Creators | Woven Ties | Select a Design</h4>
    {type && subCategories.length ? (
      <ul className="images-list">
        {subCategories.map((subCategory: ISubcategory) => (
          <SubCategory
            setType={setType}
            subCategory={subCategory}
            key={subCategory.id}
            type={type}
          />
        ))}
      </ul>
    ) : (
      <View type={type} />
    )}
  </div>
);

export default SubCategories;
