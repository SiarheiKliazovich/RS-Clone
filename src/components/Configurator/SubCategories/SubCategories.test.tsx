import { render } from "@testing-library/react";
import { MemoryRouter as Router, Route, Routes } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import SubCategories from ".";

const mockStore = configureStore([]);

describe("SubCategories", () => {
  // eslint-disable-next-line
  let store: any;

  beforeEach(() => {
    store = mockStore({
      auth: {
        user: {
          role: "USER",
        },
      },
      appInterface: {
        accentColor: {
          static: "#000",
          hover: "#fff",
        },
      },
      langReducer: {
        lang: "en",
      },
    });
  });

  test("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <Router initialEntries={["/configurator/patterns"]}>
          <Routes>
            <Route path="/category/:category" element={<SubCategories />} />
          </Routes>
        </Router>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
