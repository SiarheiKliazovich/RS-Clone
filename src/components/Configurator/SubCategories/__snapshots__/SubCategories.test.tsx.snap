// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubCategories renders correctly 1`] = `null`;
